### PR TITLE
Issue #12568: Fix lint GitHub Actions workflows executed twice

### PR DESCRIPTION
**Pull request by nrmancuso**
#12568

### CODE REVIEW

File: .github/workflows/actionlint.yml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 name: Lint GitHub Actions workflows
-on: [ push, pull_request ]
+on:
+  pull_request:
+    branches: [ master ]
```
Comment (romani) [inline .github/workflows/actionlint.yml:4]: How it will work for detect push to master (no PR) and validation after merge to master?

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: .github/workflows/actionlint.yml (hunk 1 above)
Comment (Vyom-Yadav) [reply .github/workflows/actionlint.yml:4]:
```yaml
on:
  push:
    branches:
      - master
  pull_request:
    branches: '*'
```

Comment (romani) [reply .github/workflows/actionlint.yml:4]: @nrmancuso , ∆∆

Comment (nrmancuso) [reply .github/workflows/actionlint.yml]: Done, additionally identified a couple of others that needed the same config.

**Comment by romani** [comment]
@nrmancuso , after merge please test it by PR from branch in this repo (not a fork)

**Review by romani — APPROVED** [review]

**Review by rnveach — APPROVED** [review]

**Review by Vyom-Yadav — APPROVED** [review]

**Comment by nrmancuso** [comment]
All CI failures are known or random.

**Pull request merged**